skimage/segmentation/_seeds.pyx

```diff
@@ -0,0 +1,725 @@
+"""
+Helper module for implementation of the SEEDS segmentation algorithm [1]_.
+
+Written by Geoffrey French 2015 and released under BSD license.
+
+
+References
+----------
+
+.. [1] M. van den Bergh, X. Boix, G. Roig, B. de Capitani, L. van Gool..
+       "SEEDS: Superpixels extracted via energy-driven sampling."
+       Proceedings of the European Conference on Computer Vision, pages 13-26,
+       2012.
+"""
+import cython
+import numpy as np
+
+
+
+cdef class LevelMetrics:
+    """
+    Describes the downscaling that was performed to get to the level described
+    by `self`
+
+    Attributes
+    ----------
+    h, w : the height and width of the original image
+    block_h, block_w : the height and width of a block in the downsampled image
+    last_block_h, last_block_w : the height and width of blocks in the last
+    row and column respectively; used in cases where the `block_h` and
+    `block_w` do not divide exactly into `h` and `w`
+    rem_y, rem_x : the remainder of dividing `h` and `h` by `block_h` and
+    `block_w` respectively
+    n_blocks_y, n_blocks_x : the number of `(block_h,block_w)` blocks that
+    fit into the `(h,w)` image
+
+    """
+    cdef readonly int h, w
+    cdef readonly int block_h, block_w
+    cdef readonly int last_block_h, last_block_w
+    cdef readonly int rem_y, rem_x
+    cdef readonly int n_blocks_y, n_blocks_x
+
+    cpdef initialise(self, int h, int w, int block_h, int block_w):
+        """
+        Initialise, given an image and a block size
+
+        Parameters
+        ----------
+        h, w : the shape of the image
+        block_h, block_w : the shape of blocks
+        """
+        self.h = h
+        self.w = w
+        self.block_h = block_h
+        self.block_w = block_w
+        self.n_blocks_y = h // block_h
+        self.n_blocks_x = w // block_w
+        self.rem_y = h % block_h
+        self.rem_x = w % block_w
+        self.last_block_h = block_h + self.rem_y
+        self.last_block_w = block_w + self.rem_x
+
+    cpdef downscale(self, LevelMetrics metrics):
+        """
+        Initialise by downscaling an existing level by a factor of 2.
+        """
+        self.h = metrics.h
+        self.w = metrics.w
+        self.block_h = metrics.block_h * 2
+        self.block_w = metrics.block_w * 2
+        self.n_blocks_y = metrics.n_blocks_y // 2
+        self.n_blocks_x = metrics.n_blocks_x // 2
+        self.rem_y = metrics.h % self.block_h
+        self.rem_x = metrics.w % self.block_w
+        self.last_block_h = self.block_h + self.rem_y
+        self.last_block_w = self.block_w + self.rem_x
+
+    cpdef int block_area(self, int y, int x):
+        """
+        Compute the area of the block in original image pixels at the given
+        co-ordinates
+
+        Parameters
+        ----------
+        y, x: the block co-ordinates
```

Comment:
See the [NumPy docstring conventions](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)